bluemira/display/plotter.py

```diff
@@ -50,224 +50,130 @@
 if TYPE_CHECKING:
     from bluemira.geometry.base import BluemiraGeo
 
-DEFAULT_PLOT_OPTIONS = {
-    # flags to enable points, wires, and faces plot
-    "show_points": False,
-    "show_wires": True,
-    "show_faces": True,
-    # matplotlib set of options to plot points, wires, and faces. If an empty dictionary
-    # is specified, the default color plot of matplotlib is used.
-    "point_options": {"s": 10, "facecolors": "red", "edgecolors": "black", "zorder": 30},
-    "wire_options": {"color": "black", "linewidth": 0.5, "zorder": 20},
-    "face_options": {"color": "blue", "zorder": 10},
-    # reference plotting placement
-    "view": "xz",
-    # discretization properties for plotting wires (and faces)
-    "ndiscr": 100,
-    "byedges": True,
-}
-
 UNIT_LABEL = "[m]"
 X_LABEL = f"x {UNIT_LABEL}"
 Y_LABEL = f"y {UNIT_LABEL}"
 Z_LABEL = f"z {UNIT_LABEL}"
 
 
-def get_default_options():
-    """
-    Returns the instance as a dictionary.
-    """
-    output_dict = {}
-    for k, v in DEFAULT_PLOT_OPTIONS.items():
-        # FreeCAD Placement that is contained in BluemiraPlacement cannot be
-        # deepcopied by copy.deepcopy. For this reason, its deepcopy method must be
-        # used in this case.
-        if isinstance(v, _placement.BluemiraPlacement):
-            output_dict[k] = v.deepcopy()
-        else:
-            output_dict[k] = copy.deepcopy(v)
-    return output_dict
+class ViewDescriptor:
+    """Descriptor for placements in dataclass"""
 
+    def __init__(self):
+        self._default = tuple(
+            getattr(_placement.XZY, attr) for attr in ("base", "axis", "angle", "label")
+        )
 
-class DisplayOptions:
-    """
-    The options that are available for displaying objects.
-    """
-
-    _options = None
+    def __set_name__(self, _, name: str):
+        """Set the attribute name from a dataclass"""
+        self._name = "_" + name
+
+    def __get__(
+        self, obj: Any, _
+    ) -> Tuple[Iterable[float], Iterable[float], float, str]:
+        """Get the view tuple"""
+        if obj is None:
+            return self._default
+
+        return getattr(obj, self._name, self._default)
+
+    def __set__(self, obj: Any, value: Union[str, _placement.BluemiraPlacement]):
+        """Set the view"""
+        if isinstance(value, str):
+            if value == "xy":
+                value = _placement.XYZ
+            elif value == "xz":
+                value = _placement.XZY
+            elif value == "yz":
+                value = _placement.YZX
```

Comment:
Maybe I'm missing something obvious, but why are the strings here different from the keys in the `BluemiraPlacement` enum?

Reply:
I'm not sure I follow. `_placement.XYZ` is a `BluemiraPlacement` defined in the `bluemira.geometry.placement` module

Reply:
for some reason the import previously was from `bluemira.geometry import placement as _placement` unsure why but I kept it

Reply:
discussed elsewhere